### DISCUSSION TITLE
UWB tracking questions

### DISCUSSION THREAD

**Discussion by kalib-loop** [bitcraze/discussions#372] [Q&A]
I have some questions regarding capability of the different tracking methods compatible with the Crazyflie drones.

Starting with UWB 
When using TDOA3 what range have people gotten with the Crazyflies? and what is the highest amount of drones people have gotten flying at one time? when flying in large swarms what are the biggest issues you encountered aside from space?

When using TDOA3 on the long range setting the documentation claims a 40M range but at a reduced bit rate. how drastically does this affect the accuracy of the tracking? has this caused issues or is it not noticeable? 

Aside form the study sited in the documentation on the Bitcraze site has anyone used the (DS-TWR) protocol and kind of tracking results did it provide?

Using UWB deck with LED ring deck what kind of flight time are people getting before having to recharge?

Are the charging stations featured in Bitcraze https://www.bitcraze.io/wp-content/uploads/2022/10/booth-set-up.png video from japan purchasable? I was not able to find them on your site.

**Comment by krichardsson** [comment]
> When using TDOA3 what range have people gotten with the Crazyflies? 

TDoa3 in the default mode (smart TX power) probably reaches around 3-4 meters. With TX full power 8-10 meters.

> and what is the highest amount of drones people have gotten flying at one time? when flying in large swarms what are the biggest issues you encountered aside from space?

From a positioning point of view, I don't see that there should be any limitations on the number of Crazyflies. The bigger problems are usually to fit many drones in a small space without colliding or being hit by down-wash. The other big problem is to make sure all drones in the fleet are "good", that is has a charged battery, has good radio communication, propellers and motors are OK, decks are mounted properly and so on.

> When using TDOA3 on the long range setting the documentation claims a 40M range but at a reduced bit rate. how drastically does this affect the accuracy of the tracking? has this caused issues or is it not noticeable?

We don't have a lot of experience in this area, but I don't think the difference of accuracy in tracking is noticeable.

> Aside form the study sited in the documentation on the Bitcraze site has anyone used the (DS-TWR) protocol and kind of tracking results did it provide?

I'm not sure what you are referring to as the DS-TWR, do you have a link to the documenation?

> Using UWB deck with LED ring deck what kind of flight time are people getting before having to recharge?

With a LED ring and a Loco deck it can probably hover for about 5 minutes. With more motions the stable flight time is maybe around 3-4 minutes. If the batteries are worn out even less.

> Are the charging stations featured in Bitcraze https://www.bitcraze.io/wp-content/uploads/2022/10/booth-set-up.png video from japan purchasable? I was not able to find them on your site.

The plastic ring around is 3D printed, you can find it here https://github.com/bitcraze/bitcraze-mechanics/blob/master/models/Landing_pad_v5.stl
It was originally designed for the "Rallen" chargers from IKEA, but I think "Nordmärke" will fit as well.
We don't sell the 3D printed part or the charger in our store, but it should be possible to order on-line.

Note also that it is not possible to use the LED-ring and Qi-charger deck at the same time

**Reply by kalib-loop** [reply]
Thank you very much this was extremely helpful 

For the DS-TWR ranging I found it at https://www.bitcraze.io/category/loco-positioning/ 
This talks about a study experimenting with DS-TWR but when looking at that directly did not answer my questions either.

**Reply by krichardsson** [reply]
Ah, now I understand the DS-TWR!
This was done in a research project and is not part of the firmware, but I think the source code is available if you want to try it. We have been discussing adding the same type of functionality (I usually call it TDoA3 hybrid mode) to the firmware but it has not been done yet. It would make it possible to do TWR between drones or to selected anchors from time to time. There are a couple of other research projects that have used this type of functionality as well. I would expect the accuracy to be similar to normal TWR with the extra uncertainty that the position of the other drone is not as precise as for an anchor.

**Reply by kalib-loop** [reply]
Got it, thank you :). 

Another question based on a answers you gave previously. with the LED deck and the Loco tracking Deck the crazyflie only gets 3-5 minute flight time. If I want to get that flight time up I assume I can use a larger battery but that comes at the cost of added weight and consequently reduced flight performance. Do you think that adding upgraded motors to the crazyflie as shown in post (https://www.bitcraze.io/2022/10/thrust-upgrade-kit-for-the-crazyflie-2-1/) would help compensate for this issue? Of coarse these motors would come with greater power consumption. My hope is to get the flight time up without a significant reduction in performance. Hopefully that made sense. Also is the a place where I can purchase these motors?

Thank you very much for all your help so far. I'm sure I will have a lot more questions down the road.

**Reply by krichardsson** [reply]
Hi!

We're trying to keep one topic per discussion to make it easier for other users to find information. This question is sort of sliding away from UWB tracking, would it be possible for you to start a new discussion about the flight time? I think it is a question that other would be interested in as well and a clear title will help them find it. Thanks!

**Reply by kalib-loop** [reply]
Sure thing
